zerver/lib/upload.py

```diff
@@ -859,32 +844,26 @@ def get_realm_logo_url(self, realm_id: int, version: int, night: bool) -> str:
             file_name = "logo.png"
         return f"/user_avatars/{realm_id}/realm/{file_name}?version={version}"
 
-    def ensure_medium_avatar_image(self, user_profile: UserProfile) -> None:
+    def ensure_avatar_image(
+        self, user_profile: UserProfile, is_medium: bool = False
+    ) -> None:  # nocoverage
+        file_extension = "-medium.png" if is_medium else ".png"
         file_path = user_avatar_path(user_profile)
 
-        output_path = os.path.join(settings.LOCAL_UPLOADS_DIR, "avatars", file_path + "-medium.png")
-        if os.path.isfile(output_path):
-            return
-
-        image_path = os.path.join(settings.LOCAL_UPLOADS_DIR, "avatars", file_path + ".original")
-        with open(image_path, "rb") as f:
-            image_data = f.read()
-        resized_medium = resize_avatar(image_data, MEDIUM_AVATAR_SIZE)
-        write_local_file("avatars", file_path + "-medium.png", resized_medium)
-
-    def ensure_basic_avatar_image(self, user_profile: UserProfile) -> None:  # nocoverage
-        # TODO: Refactor this to share code with ensure_medium_avatar_image
-        file_path = user_avatar_path(user_profile)
-
-        output_path = os.path.join(settings.LOCAL_UPLOADS_DIR, "avatars", file_path + ".png")
+        output_path = os.path.join(
+            settings.LOCAL_UPLOADS_DIR, "avatars", file_path + file_extension
+        )
         if os.path.isfile(output_path):
             return
 
         image_path = os.path.join(settings.LOCAL_UPLOADS_DIR, "avatars", file_path + ".original")
         with open(image_path, "rb") as f:
             image_data = f.read()
-        resized_avatar = resize_avatar(image_data)
-        write_local_file("avatars", file_path + ".png", resized_avatar)
+        if is_medium:
+            resized_medium = resize_avatar(image_data, MEDIUM_AVATAR_SIZE)
+        else:
+            resized_medium = resize_avatar(image_data)
+        write_local_file("avatars", file_path + file_extension, resized_medium)
```

Comment:
`resized_avatar` would be a better name for this variable in the combined code path, since it's not always medium-size but is always resized.